scripts/speech_recognition/estimate_data_weights.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from contextlib import contextmanager
+from typing import Sequence
+
+import click
+import numpy as np
+from omegaconf import DictConfig, ListConfig, OmegaConf
+
+from nemo.collections.common.data.lhotse.cutset import get_parser_fn
+
+
+@click.command()
+@click.argument("input_cfgs", type=click.Path(exists=True, dir_okay=False), nargs=-1)
+@click.argument("output_cfg", type=click.Path())
+@click.option(
+    "-t",
+    "--temperature",
+    type=float,
+    default=None,
+    multiple=True,
+    help="Temperature for re-weighting datasets. 1 is a neutral value. "
+    "Lower temperature over-samples smaller datasets, and vice versa. "
+    "Can be specified multiple times to apply a different temperature to each group level in the YAML config.",
```

Comment:
Could you specify example of nesting temp? (Group levels in the YAML config can be counterintuitive with current setup.) Should be clear if it's from highest level to lowest level or vice versa. May also make life easier if we can pass weight paired with group name. `{source_lang: 0.5, task: 0.25}`

Reply:
Will add an example in the doc. 

> May also make life easier if we can pass weight paired with group name. {source_lang: 0.5, task: 0.25}

Unfortunately there is no concept of a group name in this config. We could try to re-weight based on attributes but I'm not sure the implementation will be trivial. These configs have a tree-like structure so we'll need one traversal to count the weights per attribute, determine the attribute weight multiplier, and another traversal to update the weights. + handling of cases when some dataset doesn't have a tag present. Is it worth it? (CC @nithinraok you had a similar request)

Reply:
Worth it as in run time? Tree traversal is dwarfed by the cost to iterate through manifests.

Worth it as in a pain to write? We can merge this then add the change as a follow up pr. We'll likely be suffering headaches without it so will need the change eventually.